packages-ts/gauntlet-terra-contracts/src/commands/contracts/ocr2/inspection/responses.ts

```diff
@@ -0,0 +1,200 @@
+import { RDD } from '@chainlink/gauntlet-terra'
+import { InspectInstruction, instructionToInspectCommand } from '../../../abstract/inspectionWrapper'
+import { CONTRACT_LIST } from '../../../../lib/contracts'
+import { CATEGORIES } from '../../../../lib/constants'
+import { LCDClient } from '@terra-money/terra.js'
+import { getLatestOCRNewTransmissionEvent, parseObservers, RoundData } from '../../../../lib/inspection'
+import { inspection, logger } from '@chainlink/gauntlet-core/dist/utils'
+
+type CommandInput = {
+  transmitters: string[]
+  description: string
+  operators: string[]
+  aggregatorOracles: any[]
+}
+
+type ContractExpectedInfo = {
+  transmitters: string[]
+  description: string
+  operators: string[]
+  aggregatorOracles: any[]
+  aggregatorAddress?: string
+  latestRoundData?: RoundData
+  event?: any
+}
+
+type ObservationSummary = {
+  nodeAddress: string
+  name: string
+  submission: string
+  apis: string
+}
+
+const makeInput = async (flags: any, args: string[]): Promise<CommandInput> => {
+  if (flags.input) return flags.input as CommandInput
+  const rdd = RDD.getRDD(flags.rdd)
+  const aggregatorAddress = args[0]
+  const aggregatorProps = rdd.contracts[aggregatorAddress]
+  const aggregatorOracles = aggregatorProps['oracles']
+  const aggregatorOperators: string[] = aggregatorOracles.map((o) => o.operator)
+
+  const operators = rdd.operators
+  const transmitters = aggregatorOperators.map((o) => rdd.operators[o].ocrNodeAddress[0])
+
+  return {
+    transmitters,
+    description: aggregatorProps.name,
+    operators,
+    aggregatorOracles,
+  }
+}
+
+const makeInspectionData = () => async (input: CommandInput): Promise<ContractExpectedInfo> => {
+  return {
+    transmitters: input.transmitters,
+    description: input.description,
+    operators: input.operators,
+    aggregatorOracles: input.aggregatorOracles,
+  }
+}
+
+const makeOnchainData = (provider: LCDClient) => async (
+  instructionsData: any[],
+  input: CommandInput,
+  aggregator: string,
+): Promise<ContractExpectedInfo> => {
+  const latestRoundData = instructionsData[0]
+  const transmitters = instructionsData[1]
+  const description = instructionsData[2]
+
+  const event = await getLatestOCRNewTransmissionEvent(provider, aggregator)
+
+  return {
+    transmitters: transmitters.addresses,
+    description,
+    operators: input.operators,
+    aggregatorOracles: input.aggregatorOracles,
+    aggregatorAddress: aggregator,
+    latestRoundData: {
+      roundId: latestRoundData.round_id,
+      answer: latestRoundData.answer,
+      observationsTimestamp: latestRoundData.observations_timestamp,
+      transmissionTimestamp: latestRoundData.transmission_timestamp,
+    },
+    event,
+  }
+}
+
+const inspect = (expected: ContractExpectedInfo, onchainData: ContractExpectedInfo): boolean => {
```

Comment:
Try to refactor this function using a better structured Input, as I suggested in the previous comment. 
Remember that here there's no concept of RDD anymore, just the user Input. RDD is just a source that we use to generate an Input.